google-cloud-pubsub/lib/google/cloud/pubsub/publisher.rb

```diff
@@ -0,0 +1,205 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "monitor"
+require "concurrent"
+require "google/cloud/pubsub/publish_result"
+require "google/cloud/pubsub/service"
+
+module Google
+  module Cloud
+    module Pubsub
+      ##
+      # @private
+      # # Publisher
+      #
+      class Publisher
+        include MonitorMixin
+
+        MAX_BYTES = 5*1024*1024
+        MAX_MESSAGES = 1000
+        TIMER_DELAY = 0.25
```

Comment:
The spec states:

> a Publisher is instantiated which runs on its own Thread publishing enqueued message on a fast cadence (eg. every 0.25 seconds)

To me that indicated that the Publisher manages the duration that messages are published, not the batches of messages. Should this be discussed on the specification document?